frontend/public/components/modals/taint-update-modal.tsx

```diff
@@ -0,0 +1,140 @@
+/* eslint-disable no-undef, no-unused-vars */
+
+import * as React from 'react';
+import {Dropdown, PromiseComponent} from '../utils';
+import { K8sKind, k8sPatch } from '../../module/k8s';
+
+import {createModalLauncher, ModalBody, ModalComponentProps, ModalSubmitFooter, ModalTitle} from '../factory';
+import * as _ from 'lodash-es';
+
+class TaintUpdateModal extends PromiseComponent {
+  readonly state: TaintUpdateModalState;
+
+  constructor(public props: TaintUpdateModalProps) {
+    super(props);
+    this.state.taints = this.props.resource.spec.taints || [];
+  }
+
+  _submit = (e) => {
+    e.preventDefault();
+
+    // We just throw away any rows where the key is blank
+    const taints =this.state.taints;
```

Comment:
```suggestion
    const { taints } = this.state;
```

or just reference `this.state.taints` below since you aren't using destructuring for props